web/index.js

```diff
@@ -166,10 +164,9 @@ export async function createServer(
 
   app.use("/*", async (req, res, next) => {
     const shop = req.query.shop;
+    const appInstalled = await AppInstallations.includes(shop);
 
-    // Detect whether we need to reinstall the app, any request from Shopify will
-    // include a shop in the query parameters.
-    if (app.get("active-shopify-shops")[shop] === undefined && shop) {
+    if (shop && !appInstalled) {
```

Comment:
@mkevinosullivan Won't this fail for online tokens where each user needs their own session?

e.g.
1. User 1 visits app on shop 1
    - Session record created at user-level e.g. `{ id: 'shop1.myshopify.com_user1', shop: 'shop1.myshopify.com' }`
2. User 2 visits app on shop 1
    - User won't be sent through oauth because user 1 has already installed the app so there won't be a session created for `shop1.myshopify.com_user2`?

Granted, this is the same logic as before but I think it was broken beforehand anyway? There's some related discussion here https://github.com/Shopify/shopify-app-template-node/issues/814#issuecomment-1145081525